internal/gatewayapi/testdata/envoyproxy-tracing-backend-uds.out.yaml

```diff
@@ -0,0 +1,282 @@
+                  volumes:
+                  - name: certs
+                    secret:
+                      secretName: envoy-cert
+                replicas: 2
+              envoyService:
+                type: LoadBalancer
+            type: Kubernetes
+          telemetry:
+            tracing:
+              provider:
+                backendRefs:
+                - group: gateway.envoyproxy.io
+                  kind: Backend
+                  name: datadog-svc
+                  namespace: envoy-gateway-system
+                  port: 8126
+                backendSettings:
+                  circuitBreaker:
+                    maxConnections: 2048
+                  connection:
+                    bufferLimit: 20Mi
+                  healthCheck:
+                    passive:
+                      consecutiveGatewayErrors: 4
+                      interval: 5s
+                  http2:
+                    initialConnectionWindowSize: 2Mi
+                    initialStreamWindowSize: 128Ki
+                    maxConcurrentStreams: 200
+                    onInvalidMessage: TerminateStream
+                  loadBalancer:
+                    consistentHash:
+                      header:
+                        name: X-some-header
+                      type: Header
+                    type: ConsistentHash
+                  proxyProtocol:
+                    version: V2
+                  tcpKeepalive:
+                    probes: 7
+                  timeout:
+                    tcp:
+                      connectTimeout: 15s
+                type: OpenTelemetry
+              samplingRate: 100
+        status: {}
+      listeners:
+      - address: null
+        name: envoy-gateway/gateway-1/http
+        ports:
+        - containerPort: 10080
+          name: http-80
+          protocol: HTTP
+          servicePort: 80
+      metadata:
+        labels:
+          gateway.envoyproxy.io/owning-gateway-name: gateway-1
+          gateway.envoyproxy.io/owning-gateway-namespace: envoy-gateway
+        ownerReference:
+          kind: GatewayClass
+          name: envoy-gateway-class
+      name: envoy-gateway/gateway-1
+      namespace: envoy-gateway-system
+xdsIR:
+  envoy-gateway/gateway-1:
+    accessLog:
+      json:
+      - path: /dev/stdout
+    http:
+    - address: 0.0.0.0
+      hostnames:
+      - '*'
+      isHTTP2: false
+      metadata:
+        kind: Gateway
+        name: gateway-1
+        namespace: envoy-gateway
+        sectionName: http
+      name: envoy-gateway/gateway-1/http
+      path:
+        escapedSlashesAction: UnescapeAndRedirect
+        mergeSlashes: true
+      port: 10080
+    readyListener:
+      address: 0.0.0.0
+      ipFamily: IPv4
+      path: /ready
+      port: 19003
+    tracing:
+      destination:
+        metadata:
+          kind: EnvoyProxy
+          name: test
+          namespace: envoy-gateway-system
+        name: tracing
+        settings:
+        - addressType: IP
```

Comment:
should ideally be `UDS`
looks like this doesnt impact the final translation ?

Reply:
valid values list as following:
```
// DestinationAddressType describes the address type state for a group of DestinationEndpoint
type DestinationAddressType string

const (
	IP    DestinationAddressType = "IP"
	FQDN  DestinationAddressType = "FQDN"
	MIXED DestinationAddressType = "Mixed"
)
```

Reply:
dont want to block this bug, can we track this with a  follow up ?

Reply:
https://github.com/envoyproxy/gateway/issues/6408